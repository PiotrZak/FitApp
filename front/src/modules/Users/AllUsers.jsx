
import React, { useState, useEffect, useContext } from 'react';
import { useDispatch } from 'react-redux';
import { userService } from 'services/userServices';
import { planService } from 'services/planService';
import { alertActions } from 'redux/actions/alert.actions';
import Return from 'common/Return';
import { commonUtil } from 'utils/common.util';
import { Loader } from 'common/Loader';
import Icon from 'common/Icon';
import { Link } from 'react-router-dom';
import { CheckboxGenericComponent } from 'common/CheckboxGenericComponent';
import { userContext } from 'App';
import { Button } from 'common/buttons/Button';
import InviteUserModal from './InviteUsersModal';
import { isMobile} from "react-device-detect";

var ReactBottomsheet = require('react-bottomsheet');

const usersText = "All Users";
const returnToSubMenu = "return"
const assignPlanToUserNotification = "Plan sucesfully assigned to users!";
const noUsers = "There are no Users";
const deleteUserText = "Delete user"
const assignToTrainerText = "Assign to trainer"
const assignPlanText = "Assign plan"
const selectFromPlans = "Select from plans"
const selected = "selected";
const userDeleted = "Users sucessfully deleted!"

export const AllUsers = () => {

  const { user } = useContext(userContext);
  const [users, setUsers] = useState();
  const [activeUsers, setActiveUsers] = useState([]);

  const [assignPlan, setAssignPlan] = useState(false);
  const [assignTrainer, setAssignTrainer] = useState(false);

  const [isLoading, setIsLoading] = useState(true);
  const [openInviteUserModal, setOpenInviteUserModal] = useState(false);

  const [bottomSheet, setBottomSheet] = useState(false);

  const dispatch = useDispatch();

  useEffect(() => {
    getAllUsers();
  }, []);

  const getAllUsers = () => {
    userService
      .allUsers()
      .then((data) => {
        setUsers(data);
        setIsLoading(false);
      })
      .catch((error) => {
        dispatch(alertActions.error(error.title));
      });

  };

  const submissionHandleElement = (selectedData) => {
    console.log(selectedData)
    const selectedUsers = commonUtil.getCheckedData(selectedData, 'userId');
    setActiveUsers(selectedUsers);
    console.log(selectedUsers)
    if (selectedUsers.length > 0) {
      setBottomSheet(true);
    } else {
      setBottomSheet(false);
      setAssignPlan(false);
    }
  };


  return (
    <div>
      <div className="container">
        <div className="container__title">
          <Return />
          <h2>{usersText}</h2>
          <div onClick={() => setOpenInviteUserModal(true)}><Icon name="plus" fill="#5e4ae3" /></div>
        </div>
        <div className="users">
          <div className="users__filters">
            <Link to={{
              pathname: `/trainers`,
            }}><p>Trainers page</p></Link>
            <Link to={{
              pathname: `/clients`,
            }}><p>Clients page</p></Link>
            <hr />
          </div>
          <InviteUserModal openModal={openInviteUserModal} onClose={() => setOpenInviteUserModal(false)} />
          <Loader isLoading={isLoading}>
            {users ? <CheckboxGenericComponent dataType="users" displayedValue="firstName" dataList={users} onSelect={submissionHandleElement} /> : <h1>{noUsers}</h1>}
          </Loader>
        </div>
      </div>
      <UsersPanel bottomSheet={bottomSheet} setBottomSheet={setBottomSheet} activeUsers = {activeUsers} setAssignPlan ={setAssignPlan} setAssignTrainer ={setAssignTrainer}/>
      <AssignUsersToPlans assignPlan={assignPlan} setAssignPlan={setAssignPlan} bottomSheet={bottomSheet} setBottomSheet={setBottomSheet} activeUsers={activeUsers} />
      <AssignUsersToTrainers assignTrainer={assignTrainer} setAssignTrainer={setAssignTrainer} bottomSheet={bottomSheet} setBottomSheet={setBottomSheet} activeUsers={activeUsers} />
    </div>
  );
};





export const UsersPanel = ({bottomSheet, setBottomSheet, activeUsers, setAssignPlan, setAssignTrainer}) => {

  const dispatch = useDispatch();

  useEffect(() => {
    if(activeUsers == 0){
      setBottomSheet(false)
    }
  }, [activeUsers]);

  const deleteUser = () => {
    userService
      .deleteUsers(activeUsers)
      .then((data) => {
        dispatch(alertActions.success(userDeleted));
      })
      .catch((error) => {
        dispatch(alertActions.error(error.title));
      });
  };

  const openAssignPlansToUsers = () => {
    setAssignPlan(true);
    setBottomSheet(false);
  };

  const openAssignTrainersToUsers = () => {
    setAssignTrainer(true);
    setBottomSheet(false);
  };

  return(
    <ReactBottomsheet
    showBlockLayer={false}
    className={!isMobile && "bottomsheet-without-background"}
    visible={bottomSheet}
    onClose={() => setBottomSheet(false)}
    appendCancelBtn={false}
  >

    {isMobile ?
      <>
        <button onClick={() => deleteUser()} className="bottom-sheet-item">{deleteUserText}</button>
        <button onClick={() => openAssignPlansToUsers()} className="bottom-sheet-item">{assignPlanText}</button>
        <button onClick={() => openAssignTrainersToUsers()} className="bottom-sheet-item">{assignToTrainerText}</button>
      </>
      :
      <>
        <div className="bottom-sheet-item__oneline">
          <Icon name="check" fill="#2E6D2C" />
          <p>{activeUsers.length} {selected}</p>
          <div onClick={() => deleteUser()} className="bottom-sheet-item__content"><Icon height={"18px"} name="trash" fill="#C3C3CF" />{deleteUserText}</div>
          <div onClick={() => openAssignPlansToUsers()} className="bottom-sheet-item__content"><Icon height={"18px"} name="clipboard-notes" fill="#C3C3CF" />{assignPlanText}</div>
          <div onClick={() => openAssignTrainersToUsers()} className="bottom-sheet-item__content"><Icon height={"18px"} name="user-circle" fill="#C3C3CF" />{assignToTrainerText}</div>
        </div>
      </>
    }
  </ReactBottomsheet>
  )
}


export const AssignUsersToPlans = ({
  assignPlan, setAssignPlan, activeUsers, setBottomSheet,
}) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [plans, setPlans] = useState();
  const [activePlans, setActivePlans] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const dispatch = useDispatch();

  useEffect(() => {
    getAllPlans();
      if(activeUsers == 0){
        setAssignPlan(false)
      }
    }, [activeUsers]);

  const closeAssignPlansToUser = () => {
    setBottomSheet(true);
    setAssignPlan(false);
  };

  const filterPlans = (event) => {
    setSearchTerm(event.target.value);
  };

  const plansResults = !searchTerm
    ? plans
    : plans.filter((plan) => plan.name.toLowerCase().includes(searchTerm.toLocaleLowerCase()));

  const getAllPlans = () => {
    planService
      .getAllPlans()
      .then((data) => {
        setPlans(data);
        setIsLoading(false);
      })
      .catch(() => {
      });
  };

  const getSelectedPlanIds = (selectedData) => {
    const selectedPlans = commonUtil.getCheckedData(selectedData, 'planId');
    setActivePlans(selectedPlans);
  };

  const assignUserToPlan = () => {
    const data = { clientIds: activeUsers, planIds: activePlans };

    userService
      .assignPlanToUser(data)
      .then(() => {
        dispatch(alertActions.success(assignPlanToUserNotification));
        setAssignPlan(false);
        setBottomSheet(false);
      })
      .catch((error) => {
        dispatch(alertActions.error(error.title));
      });
  };

  return (
    <ReactBottomsheet
      showBlockLayer={false}
      className="bottomsheet-without-background"
      visible={assignPlan}
      onClose={() => assignPlan(false)}
      appendCancelBtn={false}
    >

      <div className="bottom-nav">
        <div className="bottom-nav__item">
          <Icon name="check" fill="#2E6D2C" />
          <p>
            {activeUsers.length}
            {' '}
            selected
      </p>
        </div>

        <div className="bottom-nav__item">
        <Icon name="arrow-left" fill="#5E4AE3" />
          <p onClick={() => closeAssignPlansToUser()}>
            {returnToSubMenu}
          </p>
        </div>
      </div>

      <div>
        <h4>{selectFromPlans}</h4>
        <Loader isLoading={isLoading}>
          {plansResults ? <CheckboxGenericComponent dataType="plans" displayedValue="title" dataList={plansResults} onSelect={getSelectedPlanIds} /> : <p>No Plans</p>}
        </Loader>
        <Button disabled={activePlans.length === 0} className="btn btn--primary btn--lg" onClick={assignUserToPlan} name={activePlans.length === 0 ? "Select Plan" : "Assign Plans to Users"} />
      </div>

    </ReactBottomsheet>
  );
};




export const AssignUsersToTrainers = ({
  assignTrainer, setAssignTrainer, activeUsers, setBottomSheet,
}) => {

  const [trainers, setTrainers] = useState();
  const [activeTrainers, setActiveTrainers] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const dispatch = useDispatch();

  
useEffect(() => {
  getAllTrainers();
    if(activeUsers == 0){
      setAssignTrainer(false)
    }
  }, [activeUsers]);

  const closeAssignPlansToUser = () => {
    setBottomSheet(true);
    setAssignTrainer(false);
  };

  const getAllTrainers = () => {
    userService
      .allTrainers()
      .then((data) => {
        setTrainers(data);
        setIsLoading(false);
      })
      .catch(() => {
      });
  };

  const getSelectedTrainerIds = (selectedData) => {
    const selectedTrainers = commonUtil.getCheckedData(selectedData, 'userId');
    setActiveTrainers(selectedTrainers);
  };

  const assignUserToPlan = () => {
    const data = { userIds: activeUsers, trainerIds: activeTrainers };
    userService
      .assignUsersToTrainer(data)
      .then(() => {
        dispatch(alertActions.success(assignPlanToUserNotification));
        setAssignTrainer(false)
        setBottomSheet(false);
      })
      .catch((error) => {
        dispatch(alertActions.error(error.title));
      });
  };

  return (
    <ReactBottomsheet
      showBlockLayer={false}
      className="bottomsheet-without-background"
      visible={assignTrainer}
      onClose={() => setAssignTrainer(false)}
      appendCancelBtn={false}
    >

      <div className="bottom-nav">
        <div className="bottom-nav__item">
          <Icon name="check" fill="#2E6D2C" />
          <p>{activeUsers.length}selected</p>
        </div>

        <div className="bottom-nav__item">
        <Icon name="arrow-left" fill="#5E4AE3" />
          <p onClick={() => closeAssignPlansToUser()}>
            {returnToSubMenu}
          </p>
        </div>
      </div>

      <div>
        <h4>{selectFromPlans}</h4>
        <Loader isLoading={isLoading}>
            {trainers ? <CheckboxGenericComponent dataType="users" displayedValue="firstName" dataList={trainers} onSelect={getSelectedTrainerIds} /> : <h1>{noUsers}</h1>}
          </Loader>
        <Button disabled={activeTrainers.length === 0} className="btn btn--primary btn--lg" onClick={assignUserToPlan} name={activeTrainers.length === 0 ? "Select Plan" : "Assign Trainers to Users"} />
      </div>

    </ReactBottomsheet>
  );
};
