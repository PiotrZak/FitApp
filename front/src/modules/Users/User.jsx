import React, { useState, useEffect } from 'react';
import { userService } from "services/userServices";
import Icon from 'components/atoms/Icon';
import BackTopNav from 'components/molecules/BackTopNav';
import styled from 'styled-components';
import "react-multi-carousel/lib/styles.css";
import { UserInfo } from "components/molecules/UserInfo"
import { isMobile } from "react-device-detect";
import GlobalTemplate, { Nav as NavI } from "../../templates/GlobalTemplate"
import { Navs } from '../MyProfile/MyProfile';

import { TrainerClients } from "./UserProfile/TrainerClients"
import { TrainerPlans } from "./UserProfile/TrainerPlans"
import { UserPlans } from "./UserProfile/UserPlans"
import { ClientTrainers } from "./UserProfile/ClientTrainers"

const IconWrapper = styled.div`
    margin-top: .4rem;
`;

const TabsContainer = styled.div`
    display:flex;
    width:100%;
    margin:3.2rem 0 0 0;
    text-align:center;
`;

export const User = (props) => {

    const [user, setUser] = useState([])
    const [isLoading, setIsLoading] = useState(true)
    const [activeItem, setActiveItem] = useState('TrainerClients');
    useEffect(() => {
        getUserById()
    }, []);

    const { match } = props;
    let id = match.params;

    const getUserById = () => {
        userService
            .getUserById(id.id)
            .then((data) => {
                setUser(data);
                setIsLoading(false)
            })
            .catch((error) => {
            });
    }

    let sections = [];

    if (user.role === "Trainer") {
        sections = ['Trainer Clients', 'Trainer Plans']
    }
    else {
        sections = ['User Plans', 'Client Trainers']
    }

    return (
        <GlobalTemplate>
            <NavI>
                <BackTopNav />
            </NavI>
            {user && <UserInfo user={user} />}
            <TabsContainer>
                {sections.map((title, i) => (
                    <Navs
                        title={title}
                        setActiveItem={setActiveItem}
                        activeItem={activeItem}
                    />
                ))}
            </TabsContainer>
            {user.role === "Trainer"
                ?
                <>
                    {activeItem == 'Trainer Clients' && <TrainerClients id={user.userId} />}
                    {activeItem == 'Trainer Plans' && <TrainerPlans id={user.userId} />}
                </>
                :
                <>
                    {activeItem == 'User Plans' && <UserPlans id={user.userId} />}
                    {activeItem == 'Client Trainers' && <ClientTrainers id={user.userId} />}
                </>
            }
        </GlobalTemplate>
    );
}



export default User;