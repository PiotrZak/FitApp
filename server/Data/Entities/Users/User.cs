using System;
using System.ComponentModel.DataAnnotations;

namespace WebApi.Entities
{

    public abstract class User
    {
        protected User()
        {
            UserId = Guid.NewGuid().ToString();
            Avatar = new byte[] { 0x20, 0x20, 0x20, 0x20, 0x20, 0x20 };
        }
        public string OrganizationId { get; set; }
        [Key]
        public string UserId { get; set; }
        public byte[] Avatar { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string Password { get; set; }
        public byte[] PasswordHash { get; set; }
        public byte[] PasswordSalt { get; set; }
        public string Role { get; set; }
        public string Token { get; set; }
        public string ResetToken { get; set; }
        public DateTime? ResetTokenExpires { get; set; }
        public DateTime PasswordReset { get; set; }
        public string VerificationToken { get; set; }
        public bool IsActivated { get; set; }
    };
}