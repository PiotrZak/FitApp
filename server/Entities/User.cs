using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using WebApi.Controllers.ViewModels;

namespace WebApi.Entities
{
    public class User
    {
        public User()
        {
            UserId = Guid.NewGuid().ToString();
            Avatar = new byte[] { 0x20, 0x20, 0x20, 0x20, 0x20, 0x20 };
            UsersPlans = new List<UsersPlans>();
        }

        [Key]
        public string UserId { get; set; }
        public byte[] Avatar { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public int PhoneNumber { get; set; }
        public string Password { get; set; }
        public byte[] PasswordHash { get; set; }
        public byte[] PasswordSalt { get; set; }
        public string Role { get; set; }
        public string Token { get; set; }

        public virtual ICollection<UsersPlans> UsersPlans { get; set; }


        //public List<AssignPlansToUser> PlanIds { get; set; }
        //public string TrainerId { get; set; }
        //public List<Plan> Plans { get; set; }
    }
}