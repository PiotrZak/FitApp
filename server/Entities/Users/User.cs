using System;
using Microsoft.AspNetCore.Identity;

namespace WebApi.Entities
{

    public class ApplicationUser : IdentityUser
    {
        public virtual Client Client { get; set; }
        public virtual Trainer Trainer { get; set; }
    }

    //public class AppUser : IdentityUser
    //{
    //    public DomainUser DomainUser { get; set; }
    //}

    //public abstract class DomainUser
    //{
    //    public Guid Id { get; set; }
    //    public AppUser IdentityUser { get; set; }
    //}


    //public class User
    //{
    //    public User()
    //    {
    //        UserId = Guid.NewGuid().ToString();
    //        Avatar = new byte[] { 0x20, 0x20, 0x20, 0x20, 0x20, 0x20 };
    //        UsersPlans = new List<UsersPlans>();
    //        UsersTrainers = new List<UsersTrainers>();
    //    }

    //    [Key]
    //    public string UserId { get; set; }
    //    public byte[] Avatar { get; set; }
    //    public string FirstName { get; set; }
    //    public string LastName { get; set; }
    //    public string Email { get; set; }
    //    public int PhoneNumber { get; set; }
    //    public string Password { get; set; }
    //    public byte[] PasswordHash { get; set; }
    //    public byte[] PasswordSalt { get; set; }
    //    public string Role { get; set; }
    //    public string Token { get; set; }

    //    public virtual ICollection<UsersPlans> UsersPlans { get; set; }
    //    public virtual ICollection<UsersTrainers> UsersTrainers { get; set; }
    //}
}