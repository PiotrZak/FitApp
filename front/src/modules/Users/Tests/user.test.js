import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

import { AllUsers } from '../AllUsers';
import { User } from '../User';

const currentOwner = {
    avatar: null,
    email: "owner3@eventbrite.com",
    firstName: "Owner3",
    lastName: "lol",
    organizationId: "O3",
    role: "Owner",
    token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1bmlxdWVfbmFtZSI6Im93bmVyMyIsInJvbGUiOiJPd25lciIsIm5iZiI6MTYwMTEzNTMyNSwiZXhwIjoxNjAxNzQwMTI1LCJpYXQiOjE2MDExMzUzMjV9.imZhUNrfj7hHfNOdO7QJDJ8-U6Byu1JJbxh5N6nSsRY",
    userId: "owner3",
}

configure({ adapter: new Adapter() });

describe("AllUsers Page ", () => {

    const mockStore = configureStore()
    let store = mockStore()

    it('renders correctly', () => {
        const tree = renderer
          .create(<BrowserRouter history={history}><Provider store={store}><AllUsers/></Provider></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

    it('should render', () => {
        const wrapper = mount(<BrowserRouter><Provider store={store}><AllUsers/></Provider></BrowserRouter>)
        expect(wrapper.exists()).toBeTruthy()
    })
});

describe("UserPage ", () => {

    const mockStore = configureStore()
    let store = mockStore()

    const defaultProps = {
        match: { params: { id: currentOwner.userId } },
    };

    it('renders correctly', () => {
        const tree = renderer
          .create(<BrowserRouter history={history}><Provider store={store}><User {...defaultProps} /></Provider></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

    it('should render', () => {
        const wrapper = mount(<BrowserRouter history={history}><Provider store={store}><User {...defaultProps} /></Provider></BrowserRouter>)
        expect(wrapper.exists()).toBeTruthy()
    })

    it('should render bottomSheet', () => {
        const wrapper = mount(<BrowserRouter history={history}><Provider store={store}><User {...defaultProps} /></Provider></BrowserRouter>)
        expect(wrapper.exists()).toBeTruthy()
    })
});