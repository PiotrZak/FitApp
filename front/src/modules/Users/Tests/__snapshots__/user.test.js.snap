// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllUsers Page  renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="container__title"
    >
      <div
        alt="Go back"
        onClick={[Function]}
      >
        <div
          fill="#5E4AE3"
        />
      </div>
      <h2>
        Users
      </h2>
      <div
        onClick={[Function]}
      >
        <div
          fill="#5e4ae3"
        />
      </div>
    </div>
    <div
      className="users"
    >
      <div
        className="users__filters"
      >
        <a
          aria-current={null}
          href="/trainers"
          onClick={[Function]}
        >
          <p>
            Trainers page
          </p>
        </a>
        <a
          aria-current={null}
          href="/clients"
          onClick={[Function]}
        >
          <p>
            Clients page
          </p>
        </a>
        <hr />
      </div>
      <div />
      <div
        className="container"
      >
        <div
          className="justify-content-sm-center undefined row"
        >
          <div
            className="col"
          >
            <div
              className="spinner-large undefined spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bottom-sheet-wrapper bottomsheet-without-background leave hide"
  >
    <div
      className="bottom-sheet"
    >
      <div
        className="bottom-sheet-item__oneline"
      >
        <div
          fill="#2E6D2C"
        />
        <p>
          0
           
          selected
        </p>
        <div
          className="bottom-sheet-item__content"
          onClick={[Function]}
        >
          <div
            fill="#C3C3CF"
          />
          Delete user
        </div>
        <div
          className="bottom-sheet-item__content"
          onClick={[Function]}
        >
          <div
            fill="#C3C3CF"
          />
          Assign plan
        </div>
        <div
          className="bottom-sheet-item__content"
          onClick={[Function]}
        >
          <div
            fill="#C3C3CF"
          />
          Assign to trainer
        </div>
        <div
          className="bottom-sheet-item__content"
          onClick={[Function]}
        >
          <div
            fill="#C3C3CF"
          />
          Edit User Role
        </div>
      </div>
      <div />
    </div>
  </div>
  <div
    className="bottom-sheet-wrapper bottomsheet-without-background leave hide"
  >
    <div
      className="bottom-sheet"
    >
      <div
        className="bottom-nav"
      >
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#2E6D2C"
          />
          <p>
            0
             
            selected
          </p>
        </div>
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#5E4AE3"
          />
          <p
            onClick={[Function]}
          >
            return
          </p>
        </div>
      </div>
      <div>
        <h4>
          Select from plans
        </h4>
        <div
          className="container"
        >
          <div
            className="justify-content-sm-center undefined row"
          >
            <div
              className="col"
            >
              <div
                className="spinner-large undefined spinner-border text-primary"
                role="status"
              >
                <span
                  className="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
        <button
          className="btn btn--primary btn--lg"
          disabled={true}
          onClick={[Function]}
        >
          Select Plan
        </button>
      </div>
    </div>
  </div>
  <div
    className="bottom-sheet-wrapper bottomsheet-without-background leave hide"
  >
    <div
      className="bottom-sheet"
    >
      <div
        className="bottom-nav"
      >
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#2E6D2C"
          />
          <p>
            0
            selected
          </p>
        </div>
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#5E4AE3"
          />
          <p
            onClick={[Function]}
          >
            return
          </p>
        </div>
      </div>
      <div>
        <h4>
          Select from plans
        </h4>
        <div
          className="container"
        >
          <div
            className="justify-content-sm-center undefined row"
          >
            <div
              className="col"
            >
              <div
                className="spinner-large undefined spinner-border text-primary"
                role="status"
              >
                <span
                  className="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
        <button
          className="btn btn--primary btn--lg"
          disabled={true}
          onClick={[Function]}
        >
          Select Plan
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserPage  renders correctly 1`] = `
<div>
  <div
    className="user-container"
  >
    <div
      className="container"
    >
      <div
        className="user-container__container__title"
      >
        <div
          alt="Go back"
          onClick={[Function]}
        >
          <div
            fill="white"
          />
        </div>
        <div
          onClick={[Function]}
        >
          <div
            fill="white"
          />
        </div>
      </div>
      <div
        className="user-container__info"
      >
        <div>
          <div
            className="avatar false"
            onClick={[Function]}
            onDrop={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <input
              className="file-input"
              multiple={true}
              onChange={[Function]}
              type="file"
            />
          </div>
          <h2>
             
          </h2>
          <p />
          <p />
          <p />
        </div>
      </div>
      <div
        className="user-container__tabs"
      >
        <ul
          className="nav nav-tabs"
        >
          <li
            className="nav-item"
          >
            <a
              className="active nav-link"
              onClick={[Function]}
            >
              <h2>
                Plans
              </h2>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              onClick={[Function]}
            >
              <h2>
                Trainers
              </h2>
            </a>
          </li>
        </ul>
        <div
          className="tab-content"
        >
          <div
            className="tab-pane active"
          >
            <h1>
              <div>
                <h2>
                  There isn't plans
                </h2>
              </div>
            </h1>
          </div>
          <div
            className="tab-pane"
          >
            <h1>
              <div>
                <h2>
                  There isn't trainers
                </h2>
              </div>
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bottom-sheet-wrapper bottomsheet-without-background leave hide"
  >
    <div
      className="bottom-sheet"
    >
      <div
        className="bottom-sheet-item__oneline"
      >
        <div
          className="bottom-sheet-item__content"
          onClick={[Function]}
        >
          <div
            fill="#C3C3CF"
          />
          Assign plan
        </div>
        <button
          className="bottom-sheet-item"
          onClick={[Function]}
        >
          Assign to me
        </button>
      </div>
    </div>
  </div>
  <div
    className="bottom-sheet-wrapper bottomsheet-without-background leave hide"
  >
    <div
      className="bottom-sheet"
    >
      <div
        className="bottom-nav"
      >
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#2E6D2C"
          />
          <p>
            1
             
            selected
          </p>
        </div>
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#5E4AE3"
          />
          <p
            onClick={[Function]}
          >
            return
          </p>
        </div>
      </div>
      <div>
        <h4>
          Select from plans
        </h4>
        <div
          className="container"
        >
          <div
            className="justify-content-sm-center undefined row"
          >
            <div
              className="col"
            >
              <div
                className="spinner-large undefined spinner-border text-primary"
                role="status"
              >
                <span
                  className="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
        <button
          className="btn btn--primary btn--lg"
          disabled={true}
          onClick={[Function]}
        >
          Select Plan
        </button>
      </div>
    </div>
  </div>
  <div
    className="bottom-sheet-wrapper bottomsheet-without-background leave hide"
  >
    <div
      className="bottom-sheet"
    >
      <div
        className="bottom-nav"
      >
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#2E6D2C"
          />
          <p>
            1
            selected
          </p>
        </div>
        <div
          className="bottom-nav__item"
        >
          <div
            fill="#5E4AE3"
          />
          <p
            onClick={[Function]}
          >
            return
          </p>
        </div>
      </div>
      <div>
        <h4>
          Select from plans
        </h4>
        <div
          className="container"
        >
          <div
            className="justify-content-sm-center undefined row"
          >
            <div
              className="col"
            >
              <div
                className="spinner-large undefined spinner-border text-primary"
                role="status"
              >
                <span
                  className="sr-only"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
        <button
          className="btn btn--primary btn--lg"
          disabled={true}
          onClick={[Function]}
        >
          Select Plan
        </button>
      </div>
    </div>
  </div>
</div>
`;
